spec/helpers.lua

```diff
@@ -892,6 +892,33 @@ local function wait_for_invalidation(key, timeout)
   end, timeout)
 end
 
+--- Waits for the termination of a pid.
+-- @param pid_path Filename of the pid file.
+-- @param timeout (optional) in seconds, defaults to 10.
+local function wait_pid(pid_path, timeout)
+  local pid
+  local fd = io.open(pid_path)
+  if fd then
+    pid = fd:read("*l")
+    fd:close()
+  end
```

Comment:
Any reason for not using `kong.cmd.utils.kill`? It also has an `is_running()` method available that this function could make use of.

Actually, don't you think this function should live in this `kill` module instead? We could (later) make use of it in `kong stop`/`kong quit` as well.

Reply:
> Any reason for not using kong.cmd.utils.kill? 

It was required locally in the other function that was marked `only use in...`, so I assumed it wasn't desired to fetch this module here. Plus, why load a module do something that could be achieved in one line? I don't need the extra checks and logs of the `kill` module there.

> It also has an is_running() method available that this function could make use of.

Same rationale as the above, plus it uses `kill -0`. `kill` is implemented as a builtin in some shells and I've been too bitten before in incompatibilities in builtins replacing commands, so I figured this would be more reliable.

> Actually, don't you think this function should live in this kill module instead?

Yes, it could!

Reply:
> do something that could be achieved in one line?

The `kill` module does all of the pid file reading and command building, which is more than "one line". It also logs to `debug` the attempts it makes and the commands it runs, which could also turn very helpful one day. This is what I meant. Replacing `is_running()`'s `kill -0` with `ps -p` was also on the table when I suggested this, but it definitely wasn't my concern here. Not duplicating the pid file reading logic was - if that makes sense?

Reply:
I explicitly didn't want to rely my loop on a function that reads the pid file. The very thing I'm doing here is making sure that the process is really gone, and I don't want to get a false positive if the pid file has already been deleted but the process is still lingering.

Reply:
@hishamhm Gotcha. Sounds good!